src/features/device-provisioning/register.ts

```diff
@@ -69,6 +74,30 @@ export const register: RequestHandler = async (req, res) => {
 			if (device == null) {
 				throw new Error('Failed to create device');
 			}
+			if (req.apiKey?.permissions != null) {
+				// Add the permission to assign the created device to the current application for application/provisioning keys, will have no effect for device/user keys
+				req.apiKey.permissions.push(
+					`resin.device__belongs_to__application.create?device eq ${device.id} and belongs_to__application/any(a:a/actor eq @__ACTOR_ID)`,
+					`resin.application.read?actor eq @__ACTOR_ID`,
```

Comment:
These seem identical to above -- can you please move them someplace where both hooks can use them?

Reply:
Technically I think it's only needed in the other places with changes I made to it, but it will have to move here later - I've removed it now and with true multi-app it'll have to be reinstated